src/prof_data.c

```diff
@@ -939,13 +944,82 @@ prof_open_maps(const char *format, ...) {
 static bool
 prof_dump_maps(bool propagate_err) {
 	bool ret;
-	int mfd;
+	int mfd = -1;
 
 	cassert(config_prof);
 #ifdef __FreeBSD__
-	mfd = prof_open_maps("/proc/curproc/map");
-#elif defined(_WIN32)
-	mfd = -1; // Not implemented
+	int mib[4];
+	size_t sz;
+	char *buf;
+
+	mib[0] = CTL_KERN;
+	mib[1] = KERN_PROC;
+	mib[2] = KERN_PROC_VMMAP;
+	mib[3] = prof_getpid();
+	sz = 0;
+
+	if (prof_dump_write(propagate_err, "\nMAPPED_LIBRARIES:\n") &&
+	    propagate_err) {
+		ret = true;
+		goto label_return;
+	}
+
+	if (sysctl(mib, 4, NULL, &sz, NULL, 0) == -1) {
+		ret = true;
+		goto label_return;
+	}
+
+	/* Size for kinfo_vmentry */
+	sz = sz * 4 / 3;
```

Comment:
What's the reasoning behind this computation? Can we have a comment explaining?

Reply:
Well in fact this is to give enough room for kinfo_vmentry. FreeBSD libpvocstat thus underlyingly nvm does that too.

Reply:
But like, what property of kinfo_vmentry requires this tweak to the reported size? Is it just to handle races in which a mapping is added in between the two sysctl calls?

Reply:
All FreeBSD code consumer which list proc maps use this workflow to give enough room for enough process entries in between calls indeed but barely more documented than that.